### PR TITLE
feat: adding onPress prop to backdrop component

### PR DESCRIPTION
**Pull request by tarikfp**
## Motivation

There might be specific scenarios where some action needs to be executed on pressing backdrop component.

### CODE REVIEW

**Comment by joeyfigaro** [comment]
@tarikpnr doesn't [`pressBehavior`](https://gorhom.github.io/react-native-bottom-sheet/components/bottomsheetbackdrop#pressbehavior) handle `onPress` use cases?

**Comment by tarikfp** [comment]
> @tarikpnr doesn't [`pressBehavior`](https://gorhom.github.io/react-native-bottom-sheet/components/bottomsheetbackdrop#pressbehavior) handle `onPress` use cases?

As far as I can see that it does not support custom on press function. 

https://github.com/gorhom/react-native-bottom-sheet/blob/4a4a9dedef4f83b620a21e1a5670de0e1020a2f6/src/components/bottomSheetBackdrop/types.d.ts#L42

**Comment by tarikfp** [comment]
@gorhom Hi, any thoughts about this pr ?

**Comment by carlos-cne** [comment]
I'm also following this topic.

File: src/components/bottomSheetBackdrop/BottomSheetBackdrop.tsx [hunk 1]
```diff
@@ -52,6 +53,8 @@ const BottomSheetBackdropComponent = ({
 
   //#region callbacks
   const handleOnPress = useCallback(() => {
+    onPress?.()
```
Comment (gorhom) [inline src/components/bottomSheetBackdrop/BottomSheetBackdrop.tsx]: you need to add `onPress` to the `handleOnPress` dependencies list.

**Review by gorhom — CHANGES_REQUESTED** [review]
@tarikpnr thanks for submitting this PR, i left one comment

File: src/components/bottomSheetBackdrop/BottomSheetBackdrop.tsx (hunk 1 above)
Comment (tarikfp) [reply src/components/bottomSheetBackdrop/BottomSheetBackdrop.tsx]: @gorhom Hey gorhom, ooopss... you are right, I somehow missed it.

**Review by gorhom — APPROVED** [review]

**Pull request merged**